packages/opentelemetry-resources/src/Resource.ts

```diff
@@ -145,12 +153,26 @@ export class Resource implements IResource {
       return {
         ...this._syncAttributes,
         ...thisAsyncAttributes,
-        //Support for old resource implementation where _syncAttributes is not defined
+        // Support for old resource implementation where _syncAttributes is not defined
         ...((other as Resource)._syncAttributes ?? other.attributes),
         ...otherAsyncAttributes,
       };
     });
 
-    return new Resource(mergedSyncAttributes, mergedAttributesPromise);
+    return new Resource(mergedSyncAttributes, mergedAttributesPromise, mergedSchemaUrl);
+  }
+
+  /**
+   * Helper function to merge schema URLs. If both schema URLs are present and differ,
+   * a warning is logged and the first schema URL is prioritized.
+   */
+  private _mergeSchemaUrls(
+    schemaUrl1: string,
+    schemaUrl2: string
+  ): string {
+    if (schemaUrl1 && schemaUrl2 && schemaUrl1 !== schemaUrl2) {
+      diag.warn('Schema URLs differ. Using the original schema URL.');
+    }
```

Comment:
here we've already established that schema URL is a string (not undefined or null) though the type-system - why add null-checks? 🤔